inst/include/distributions/functors/multinomial_lpmf.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef MULTINOMIAL_LPMF
+#define MULTINOMIAL_LPMF
+
+#include "density_components_base.hpp"
+#include "../common/fims_vector.hpp"
+#include "../common/def.hpp"
+
+namespace fims_distributions
+{
+    /**
+     * Multinomial Negative Log-Likelihood
+     */
+    template <typename Type>
+    struct MultinomialLPMF : public DensityComponentBase<Type>
+    {
+        Type nll = 0.0;
+        vector<int> dims;
+        // data_indicator<tmbutils::vector<Type> , Type> keep;
+
+        MultinomialLPMF() : DistributionComponentBase<Type>()
+        {
+        }
+
+        virtual ~MultinomialLPMF() {}
+
+        virtual const Type evaluate()
+        {
+            this->nll_vec.resize(dims[0]);
+            fims::Vector<Type> observed_vector;
+            fims::Vector<Type> expected_vector;
+            observed_vector.resize(dims[1]);
+            expected_vector.resize(dims[1]);
+            for (int i = 0; i < dims[0]; i++)
+            {
+                for (int j = 0; j < dims[1]; j++)
+                {
+                    idx = (i * dims[1]) + j;
+                    observed_vector[j] = this->observed_value[idx];
+                    expected_vector[j] = this->expected_value[idx];
+                }
+                this->nll_vec[i] = dmultinom(observed_vector, expected_vector);
```

Comment:
Is dmultinom returning the likelihood or the negative likelihood? If the former, then should line 42 be -= instead of +=? Also is the log of the likelihood being calculated? If not, should like 42 be log(nll_vec[i]) instead of nll_vec[i]?